### DISCUSSION TITLE
Chat

### DISCUSSION THREAD

**Discussion by shiftrtech** [scottlamb/retina#14] [General]
Hey, I think this project is what I need to, I would like to chat with you about this, is there some discord server to talk or something similar?

Nice work.

**Comment by scottlamb** [comment]
Thank you. There's no discord server now. I'm traveling at the moment and not on a computer at a consistent time of day, so it might not be that helpful. I might try out a chat platform in a couple weeks.

**Comment by scottlamb** [comment]
We could try out github discussions now, though.

**Comment by sergiomeneses** [comment]
Hey im back, its ok on github discussions.

I have in mind make a project where we can stream multiple surveillance cameras, to a server and latter stream that video source to others devices (web, phones, etc), i think this project fits with my goals, so i want to be part of it.

I am dont have experience on video procesing or streaming, i start coding in rust six monts ago (developing a http API with actix-web), can you please guide me about what to learn (blogs, papers, examples, standars, specifications, etc)?, so i can get start contribuiting to this project (maybe start making some "good first issue" PRs).

Thanks in advance, great project.

P.S. I also did test WASI and i do think that it have so much potencial maybe we can add support for WebAssembly.

**Reply by scottlamb** [reply]
Sorry for the delay—I'm getting used to github discussions and missed your reply.

The project you have in mind sounds a lot like my [Moonfire NVR](https://github.com/scottlamb/moonfire-nvr). So yes, Retina absolutely is meant to support that kind of thing. I'd welcome your help in this repository, and also in Moonfire NVR if you want to work together on that layer.

I'm traveling with my family this week and need to feed my kids now, so I won't reply fully this second to your question about ways to get started. But I think the very first step is to just try out Retina's mp4 example and make sure it works with your cameras if you haven't already. I've gotten it to work with the models I have access to (some from Dahua, Hikvision, Reolink, Foscam, and GW Security). lucaszanella has tested it with a VStarcam model (which doesn't work yet). If you have any others, knowing how well those work or don't would be helpful.

**Reply by scottlamb** [reply]
I'm back home now with more time on my hands.

If you want to truly dive in to the specifications, I've gathered some of them [here](https://github.com/scottlamb/moonfire-nvr/wiki/Standards-and-specifications). But I have to warn you: these are authoritative references. They're enormous, dense, and *so so boring*. I never just sat down to read them all. I had an implementation goal in mind and found the parts relevant to that, in several passes.

I think the best way to get started is to find a good issue you'd like to work on. I have a few ideas there:

*   If retina doesn't work for your cameras, it'd be super helpful to diagnose and fix that, as mentioned above. I might not have access to your cameras, so this is something that might never happen if you don't do it. And I'm happy to teach you how to diagnose stuff with Wireshark, figure out fixes, etc. We can try real-time chat if that works better than bouncing comments here.
*   I have a todo list in the retina repository readme. I'll call out a few ideas:
    *   If your cameras support two-way audio, it might be fun to get that working via ONVIF audio backchannel.
    *   I would like to make retina more inviting to use, including improving the docs. They say the best way to learn is to teach; you could try adding doc strings for all the methods and fields, and I could help you refine them.
    *   I'd also like to trim retina's dependency list a bit; the webrtc stuff pulls in quite a lot. Might be worth looking at reimplementing the tiny bit of `rtcp` we need (which would fix #8) and similarly to switch to `sdp-types` rather than `sdp`.
*   But it might be more fun to work on Moonfire NVR, where you can add more high-level functionality. There are tons of gaps big and small, and I'd welcome help with them. If you fire it up, you might just see something missing that you really want, and I could help you work on it. Or I can help you find some good first issues there.
*   And I don't know how you feel about Javascript (Typescript, to be precise), but I desperately need help there. I'm more of a backend engineer than a frontend engineer, and right now that's super obvious when comparing the state of the server vs UI code.

**Reply by sergiomeneses** [reply]
Sorry for the delay, i didn't know about MoonfireNVR, but i yet not sure if is what we need, even so, i think i can help with the UI to get involve with the project at high level to latter go to low level

My next step is to buy some simple set of cameras and start testing MoonfireNVR, once i'll do it i will ping you on Github Discussions project's page.

If there is a way to get Moonfire NVR working with a public  live video resources, let me know so i can start right away.

Thanks.
